sroka/api/mysql/mysql.py

```diff
@@ -0,0 +1,52 @@
+from configparser import NoOptionError
+
+import mysql.connector
+import pandas as pd
+from retrying import retry
+
+import sroka.config.config as config
+
+
+@retry(stop_max_attempt_number=1,
+       wait_exponential_multiplier=1 * 2,
+       wait_exponential_max=1 * 2 * 2)
+def query_mysql(query, filename=None):
+    try:
+        # Set the options in a dictionary, in order to pass only the
+        # options that were provided in the configuration file to the
+        # MySQL connector. Passing empty values would trigger exceptions.
+        options = dict(
```

Comment:
That's what I was thinking too, but I didn't see an easy way to implement optional configuration using the current `config` package 🤔 Do you have an idea to suggest?

Reply:
Alright so I implemented a solution with the current config package but it's not very elegant unfortunately:

1. The `query_mysql` function tries calling `get_options_from_config`, which tries getting each value individually. Since none of them are mandatory, it passes on all exceptions. The `NoSectionError` exception is caught in `query_mysql` since it is not caught by `get_options_from_config`, which has the sole responsibility of getting each configuration variable and setting it in a dictionary.
2. The `options` dictionary is then passed to `validate_options`, which validates two things:
	1. Either the `host` option is given, or the `unix_socket` is given. If one of them is missing, the `validate_options` function prints an error message, returns false, and the `query_mysql` function stops.
	2. The `host` and `unix_socket` are not both given at the same time. This would create an ambiguous configuration since they are mutually exclusive. If this is the case, the `validate_options` function prints an error message, returns false, and the `query_mysql` function stops.

Do you think that's good enough, or should I look into the configuration parsing library to find a more elegant solution?

Reply:
Previous comment is solved here, thanks :).
Yes, I think that for now it is good enough. We should consider more elegant solution but as you said it could be handled in config part. This is not part of this PR.
Once I go through the comments I will take a look at code itself.